src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -21,35 +47,378 @@ const ActorInit En_Kakasi3_InitVars = {
+            }
+        }
+    }
+    this->actionFunc = func_80A91348;
+}
+
+void func_80A91348(EnKakasi3* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 angleTowardsLink;
+    s16 absAngleTowardsLink;
+
+    func_80A90E28(this);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->camId = -1;
+    if (func_8002F194(&this->actor, globalCtx)) {
+        if (!this->unk_194) {
+            if (this->unk_1A8 == 0) {
+                this->actionFunc = func_80A91284;
+            } else {
+                this->actionFunc = func_80A91760;
+            }
+        } else {
+            this->actionFunc = func_80A91284;
+        }
+        return;
+    }
+
+    angleTowardsLink = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+    if (!(this->actor.xzDistFromLink > 120.0f)) {
+        absAngleTowardsLink = ABS(angleTowardsLink);
+
+        if (absAngleTowardsLink < 0x4300) {
+            if (!this->unk_194) {
+
+                if (player->stateFlags2 & 0x1000000) {
+                    this->camId = func_800800F8(globalCtx, 0x8D4, -0x63, &this->actor, 0);
+                    globalCtx->msgCtx.msgMode = 0x37;
+                    this->dialogState = 5;
+                    this->unk_1B8 = 0.0f;
+                    func_8010B680(globalCtx, 0x40A4, NULL);
+                    player->stateFlags2 |= 0x800000;
+                    this->actionFunc = func_80A915B8;
+                    return;
+                } 
+                if (this->actor.xzDistFromLink < 80.0f) {
+                    player->stateFlags2 |= 0x800000;
+                }
+            } else if (gSaveContext.unk_12C5 != 0 && !this->unk_195) {
+
+                if (player->stateFlags2 & 0x1000000) {
+                    this->camId = func_800800F8(globalCtx, 0x8D4, -0x63, &this->actor, 0);
+                    globalCtx->msgCtx.msgMode = 0x37;
+                    this->dialogState = 5;
+                    this->unk_1B8 = 0.0f;
+                    func_8010B680(globalCtx, 0x40A8, NULL);
+                    player->stateFlags2 |= 0x800000;
+                    this->actionFunc = func_80A9187C;
+                    return;
+                } 
+                if (this->actor.xzDistFromLink < 80.0f) {
+                    player->stateFlags2 |= 0x800000;
+                }
+            }
+            func_8002F2CC(&this->actor, globalCtx, 100.0f);
+        }
+    }
+}
+
+void func_80A915B8(EnKakasi3* this, GlobalContext* globalCtx) {
+
+    if (func_8010BDBC(&globalCtx->msgCtx) == 5 && func_80106BC8(globalCtx)) {
+        func_80106CCC(globalCtx);
+        func_8010BD58(globalCtx, 0x2C);
+        this->actionFunc = func_80A91620;
+    }
+}
+
+void func_80A91620(EnKakasi3* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->msgCtx.unk_E3EE == 4 || (globalCtx->msgCtx.unk_E3EE >= 5 && globalCtx->msgCtx.unk_E3EE < 0xB)) {
+
+        if (globalCtx->msgCtx.msgMode == 0) {
```

Comment:
You didn't merge those ifs but the next one `if (globalCtx->msgCtx.unk_E3EE == 3 && globalCtx->msgCtx.msgMode == 0) {` is merged, probably do one or the other but not both for consistency

Reply:
you're right, I'll keep it unmerged, imo is a bit more readable this way...